addon/services/link-manager.ts

```diff
@@ -35,6 +49,34 @@ export default class LinkManagerService extends Service {
   createUILink(linkParams: LinkParams, uiParams: UILinkParams): UILink {
     return new UILink(this, { ...linkParams, ...uiParams });
   }
+
+  constructor(properties?: object) {
+    super(properties);
+
+    this.router.on('routeWillChange', this.handleRouteWillChange);
+    this.router.on('routeDidChange', this.handleRouteDidChange);
```

Comment:
Please use the util function imports for `on` and `off`: https://api.emberjs.com/ember/release/classes/@ember%2Fobject%2Fevented/methods